### PR TITLE
Cockpit: Only root can use Logs

### PR DESCRIPTION
**Pull request by stephdl**
NethServer/dev#6033

### CODE REVIEW

**Comment by gsanchietti** [comment]
It seems good to me, is it ready for review?

**Comment by stephdl** [comment]
yes it is

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    #
+    use esmith::ConfigDB;
+    
+    my $db = esmith::ConfigDB->open();
+    my $newProp = '';
+    my @Keys = split(',', $db->get_prop('cockpit.socket','delegation')|| '');
+
+    foreach (@Keys) {
+        my @props = split(':',$_);
+        my @props = grep { !/logs/ } @props;
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Regexp matches substrings. Use a full match instead:
```suggestion
        my @props = grep { $_ ne 'logs' } @props;
```

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # Remove the `logs` delegation to users  
+    #
+    use esmith::ConfigDB;
+    
+    my $db = esmith::ConfigDB->open();
+    my $newProp = '';
+    my @Keys = split(',', $db->get_prop('cockpit.socket','delegation')|| '');
+
+    foreach (@Keys) {
+        my @props = split(':',$_);
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Skip group name (first element of the list)
```suggestion
        my ($group_name, @props) = split(':',$_);
```

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        my @props = grep { !/logs/ } @props;
+        $newProp .= join(':',@props).',';
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Can we change the current loop item? (see the `join()` below)
```suggestion
        $_ = join(':', $group_name, @props);
```

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    chop $newProp;
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Is it really safe?

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    $db->set_prop('cockpit.socket','delegation',$newProp);
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]:
```suggestion
    $db->set_prop('cockpit.socket','delegation', join(',', @Keys));
```

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 4 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: if we do not remove it, former delegation are still allowed

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    my $db = esmith::ConfigDB->open();
+    my $newProp = '';
+    my @Keys = split(',', $db->get_prop('cockpit.socket','delegation')|| '');
+
+    foreach (@Keys) {
+        my ($group_name, @props) = split(':',$_);
+        my @props = grep {  $_ ne 'logs' } @props;
+        $_ = join(':', $group_name, @props);
+    }
+    $db->set_prop('cockpit.socket','delegation', join(',', @Keys));
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Are we missing the EOF?

Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: yep :'(

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 5 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 4 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 3 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 2 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 1 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+    #
+    # Remove the `logs` delegation to users
+    #
+    use esmith::ConfigDB;
+
+    my $db = esmith::ConfigDB->open();
+    my $newProp = '';
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Unused variable

```suggestion
```

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{
+    #
+    # Remove the `logs` delegation to users
+    #
+    use esmith::ConfigDB;
+
+    my $db = esmith::ConfigDB->open();
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: Do not reopen ConfigDB. Use `$DB` instead

```suggestion
```

Ref http://docs.nethserver.org/projects/nethserver-devel/en/v7/databases.html#migrate-fragments

**Comment by DavidePrincipi** [comment]
Please do not use force-push unless the PR is in draft. Force-pushes clear the review history!

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 7 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

File: root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation (hunk 8 above)
Comment (stephdl) [reply root/etc/e-smith/db/configuration/migrate/Remove_logs_delegation]: done

**Pull request merged**